### PR TITLE
Soup/ml refactor

### PR DESCRIPTION
**Pull request by Souparna-Mandal**
Refactoring the ML wrapper to allow custom logits configuration and also allowing for customisable FC and Classification Heads.

### CODE REVIEW

File: ml/models/default_models/CNNModel.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import torch
+from torch import nn
+
```
Comment (huiqiguo) [inline ml/models/default_models/CNNModel.py]: I had to add `from typing import List`

File: ml/models/loss_functions/custom_losses.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+"""Loss functions for GIK Training
+"""
+
+import torch
+from torch import nn
+from torch.nn import functional as F
+
```
Comment (huiqiguo) [inline ml/models/loss_functions/custom_losses.py]: I had to add `from typing import Optional`

**Review by Souparna-Mandal — COMMENTED** [review]
.

**Review by huiqiguo — APPROVED** [review]
Ok to merge
